### PR TITLE
Msde change fetch to add evidence

### PR DESCRIPTION
**Pull request by JasBeilin**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/45727)

## Description
Changed how fetch-incidents works, as it did not take into consideration evidence param or limit.
Also change the structure to use milliseconds and include filter as params to api, which make saving IDs unnecessary.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: In case instance is configured and the last incident is in the same second we are about to fetch, it might get duplicated as we did not consider the last incident's milliseconds.

## Must have
- [x] Tests

### CODE REVIEW

**Comment by JasBeilin** [comment]
FYI @David-BMS

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 1]
```diff
@@ -2322,22 +2336,87 @@ def add_remove_machine_tag_command(client: MsClient, args: dict):
...
+
+    latest_created_time = dateparser.parse(last_fetch_time,
+                                           settings={'RETURN_AS_TIMEZONE_AWARE': True, 'TIMEZONE': 'UTC'})
+
+    params = _get_incidents_query_params(client, fetch_evidence, last_fetch_time)
+    demisto.debug(f'getting alerts using {params=}')
+    incidents = []
+    # get_alerts:
+    try:
+        demisto.debug(f'getting alerts using {params=}')
```
Comment (DeanArbel) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2360]: Remove double debug message.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 2]
```diff
@@ -2322,22 +2336,87 @@ def add_remove_machine_tag_command(client: MsClient, args: dict):
...
+    # last alert is the newest as we ordered by it ascending
+    demisto.debug(f'got {len(incidents)} incidents from the API.')
+    last_run['last_alert_fetched_time'] = datetime.strftime(latest_created_time, TIME_FORMAT)
+    return incidents, last_run
+
+
+def _get_incidents_query_params(client, fetch_evidence, last_fetch_time):
+    filter_query = f'alertCreationTime+gt+{last_fetch_time}'
+    if client.alert_status_to_fetch:
+        statuses = argToList(client.alert_status_to_fetch)
```
Comment (DeanArbel) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2403]: you've got a typo here

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 3]
```diff
@@ -51,10 +51,19 @@ configuration:
   - High
   required: false
   type: 16
+- defaultvalue: '50'
+  display: Maximum Number Of Incidents To Fetch
```
Comment (DeanArbel) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:55]:
```suggestion
  display: Maximum Number of Incidents to Fetch
```

**Review by DeanArbel — APPROVED** [review]
Looks good. Please make sure to run `format` on all the incidentFields.

**Pull request merged**